splinepy/spline.py

```diff
@@ -1319,9 +1319,12 @@ def todict(self, tolist=False):
             should_copy = True
             # attr are either list or np.ndarray
             # prepare list if needed.
-            if isinstance(tmp_prop, np.ndarray) and tolist:
-                tmp_prop = tmp_prop.tolist()  # copies
-                should_copy = False
+            if tolist:
+                if isinstance(tmp_prop, np.ndarray):
+                    tmp_prop = tmp_prop.tolist()  # copies
+                    should_copy = False
+                if p == "knot_vectors":
+                    tmp_prop = [t.tolist() for t in tmp_prop]
```

Comment:
```suggestion
            if tolist and tmp_prop:
                if isinstance(tmp_prop, np.ndarray):
                    tmp_prop = tmp_prop.tolist()  # copies
                    should_copy = False
                if p.startswith("knot_vectors"):
                    tmp_prop = [t.tolist() for t in tmp_prop]
```
in case it is None

Reply:
This will not work, if e.g. tmp_prop is a list-like object, because in most cases it will raise ambiguity-error, I will add a "is not None"

Reply:
In later commits we could also change the default value of the get function to false